components/style/properties/helpers.mako.rs

```diff
@@ -33,6 +33,81 @@
     </%call>
 </%def>
 
+<%doc>
+    To be used in cases where we have a grammar like
+    "<thing> [ , <thing> ]*", but only support a single value
+    in servo
+</%doc>
+<%def name="gecko_autoarray_longhand(name, **kwargs)">
+    <%call expr="longhand(name, **kwargs)">
+        % if product == "gecko":
+            use cssparser::ToCss;
+            use std::fmt;
+
+            pub mod single_value {
+                use cssparser::Parser;
+                use parser::{ParserContext, ParserContextExtraData};
+                use properties::{CSSWideKeyword, DeclaredValue, Shorthand};
+                use values::computed::{TContext, ToComputedValue};
+                use values::{computed, specified};
+                ${caller.body()}
+            }
+            pub mod computed_value {
+                use super::single_value;
+                #[derive(Debug, Clone, PartialEq, HeapSizeOf)]
+                pub struct T(pub Vec<single_value::computed_value::T>);
+            }
+
+            impl ToCss for computed_value::T {
+                fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                    if self.0.len() >= 1 {
```

Comment:
This is probably a bit more tricky. In some props we might need to write `none` in the empty case, eg, animation-name.